### DISCUSSION TITLE
terrainandmovements And perfect emulation of a vanilla client?

### DISCUSSION THREAD

**Discussion by wisnoskij** [ORelio/Minecraft-Console-Client#1901] [Q&A] [answered]
Hello MCCTeam,

I am looking to use this to save electricity and processing power while AFKing on a server with cheat detection. Alternative clients and AFKing is allowed, but macroing, scripting, and defying physics is not. Specifically Hypixel Skyblock.

So my understanding is I will need to enable `terrainandmovements`. Is this the only setting I need to perfectly emulate a vanilla client? Will this perfectly emulate vanilla physics or is just close? To enable do I just `MinecraftClient.exe config.ini` with `config.ini` containing `terrainandmovements=TRUE` or 'MinecraftClient.exe --terrainandmovements=TRUE'?

But what versions does this work with? It states that this is version dependent, but not which ones work. Skyblock recommends 1.8.9 or if not then 1.16-1.17 but currently works with everything inbetween as well.

Thank you.

**Comment by Daenges** [comment, marked as answer]
So, after you first started the client, a `MinecraftClient.ini` should be created. Stop the client again, open the file with any text editor, search for [terrainandmovement](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Resources/config/MinecraftClient.ini#L35), set it to `true` and save the file.

Currently all versions up to 1.16.5 are supported. If you join in later versions it will not work. Due to the fact that Hypixel supports multiple versions above 1.16, you need to set the version manually in the config [here](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Resources/config/MinecraftClient.ini#L26) to `1.16.5`.

The terrain handling **tries** to act as natural as possible. Nevertheless, I can not guarantee that all anti-cheat systems are fine with it. You need to try it out yourself.
But in normal use cases, such as:
standing around, moving a bit and loading chunks, it should do its job without any problems.

If you want the client to appear as vanilla as possible, you can change the [brandinfo](https://github.com/MCCTeam/Minecraft-Console-Client/blob/master/MinecraftClient/Resources/config/MinecraftClient.ini#L28) additionally. Then the client will identify as `vanilla` to the server.

**Reply by ORelio** [reply]
As long as you connect and don't move, the gravity handling / falling piece of code has been designed to work closely like vanilla Minecraft. See #195 for more info on this. However, the code for moving around is more basic.
